src/frontend/query_commands.ml

```diff
@@ -512,6 +512,29 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     | Some res -> `Found res 
     | None -> `No_documentation)
 
+  | Expand_node pos -> 
+    let pos = Mpipeline.get_lexing_pos pipeline pos in
+    let typer = Mpipeline.typer_result pipeline in 
+    let node = Mtyper.node_at typer pos in
+    let check_ppx (node: Env.t * Mbrowse.node) : bool =
+      let _,n = node in
+      let has_a_deriver =
+        Browse_raw.has_attr ~name:"deriving" n in
+      let attrs = Browse_raw.node_attributes n in
+      List.exists ~f:(fun (a:Typedtree.attribute) ->
+        let loc = a.attr_loc in
+            loc.loc_start.pos_cnum <= pos.pos_cnum
+            && loc.loc_end.pos_cnum >= pos.pos_cnum
+      ) attrs && has_a_deriver
+    in 
+    let has_deriver = List.exists ~f:check_ppx node in
+    if has_deriver then 
+      let root = (List.hd (List.rev node)) in
+      let derived_nodes = Mbrowse.get_children pos root in
```

Comment:
You're using the cursor position here to decide whether an item is a derived item or not. Concretely, your logic is: Check whether the cursor position is contained in the item or not. If so, consider it a derived item (or the original item). If not, not.

When a deriver is defined with `ppxlib` that works! It works for the following reason: `ppxlib` always wraps all derived items into an
```
include struct <derived items> end <some attributes for odoc and merlin>
```
and it gives that `include` structure item the location of the whole item the deriver is attached to. So the `include` item will always contain the cursor position, when the cursor is somewhere on the deriving annotation and will therefore be detected correctly.

However, `ppx_deriving` doesn't wrap the derived items into such an `include`. So for a deriver registered via `ppx_deriving`, the situation is a bit strange. Example: If the deriver author has given the location of the deriver's name to the derived item, then the derived item will be detected as derived if the cursor position is on the deriver's name, whereas a cursor location on the `[@@deriving` won't detect it. Or worse: If the author has given the derived item the location of the type the deriver is attached to, then the derived item will never be detected.

The problem here: I don't know if we want to add `ppx_deriving` as a test dependency to `merlin-lib`. If not, we can't test the `ppx_deriving` behavior. So in that case, it might be better to add a note to the feature's doc saying that it works reliably for `ppxlib` derivers, but not for `ppx_deriving` derivers. @voodoos , what do you think about this?

Reply:
Yes the issue with `ppx_deriving` derivers as a whole is the fact that Merlin (currently) doesn't know how to traverse them, especially given that most don't respect location invariants [ppxlib location invariants](https://ocaml.org/p/ppxlib/latest/doc/good-practices.html#the-invariants) and so this functionality completely falls apart when used with any deriver written with `ppx_deriving`.